src/boot/bootctl.c

```diff
@@ -631,7 +631,9 @@ static int copy_one_file(const char *esp_path, const char *name, bool force) {
         char *p, *q;
         int r;
 
-        p = strjoina(BOOTLIBDIR "/", name);
+        p = strjoina(BOOTLIBDIR "/", name, ".signed");
+        if (access(p, F_OK) < 0)
+                p = strjoina(BOOTLIBDIR "/", name);
```

Comment:
Overall, I think this approach is OK. I get what @poettering is saying about multiple directories, but it is awkward to modify files in place, and it is also awkward to put them in a different directory. Writing a modified version to a suffixed filename is very convenient, easy to script and easy for a human to figure out. I think the general approach is OK. It is also compatible with adding support for more directories in the future, because file name suffixes are different and there is no ambiguity.

This patch works, but only as long as both .efi and .efi.signed are present. I think this is too ugly: we shouldn't depend on the presence of a file that will not be used. So please change the loop to accept `.efi.signed` too, and simply skip `.efi` if `.efi.signed` is present.

Also please squash the two commits together, it's all one change.

Reply:
thanks for the review, I'll take a look

Reply:
force pushed new version based on your suggestions